website/docs/sources.md

```diff
@@ -0,0 +1,289 @@
+# Sources
+
+A "source" in Dalec is an abstraction for fetching dependencies of a build.
+Usually this is source code but technically it could be anything.
+The source abstraction enables fetching build sources over various protocols.
+
+Some sources are considered as inherently file-based sources, like HTTP URLs or local directories.
+Other sources are considered as inherently directory-based sources, like git repositories.
+Depending on the source type, the behavior of certain things may be different, depending on the target implementation (e.g. mariner2, jammy, windows, etc).
+Sources are injected into the root path of the build environment using the name of the source.
+
+Ideally the content of a source is platform agnostic (e.g. no platform specific binaries).
+These sources are used to create source packages for the target platform, such as an srpm or a debian dsc.
+However, some source types may allow you to mount another source type in or are wrappers for other source types, like docker images or build sources respectively.
+Only the output of a top-level source is included in the build environment.
+These wrapper types (docker image, build) are useful for more advanced use-cases where you need to generate content or utilize other tooling in order to create the source.
+
+## Top-level source configuration
+
+For all source types, you can specify the following top-level configuration:
+
+- `path`: The path to extract from the source type
+- `includes`: A list of glob patterns to include from the source
+- `excludes`: A list of glob patterns to exclude from the source
+
+The below example uses a [`context`](#build-context) source type.
+The root of the source is the `path/in/source` directory.
+The source will include all `.txt` files within `path/in/source` except for `secret.txt`.
+
+```yaml
+sources:
+  someSource:
+    path: path/in/source
+    includes:
+      - "*.txt"
+    excludes:
+      - "secret.txt"
+    context: {}
+```
+
+## Source Types
+
+### Git
+
+Git sources fetch a git repository at a specific commit.
+You can use either an SSH style git URL or an HTTPS style git URL.
+
+For SSH style git URLs, if the client (such as the docker CLI) has provided
+access to an SSH agent, that agent will be used to authenticate with the git
+server.
+
+```yaml
+sources:
+  someSource1:
+    git:
+      # This uses an SSH style git URL.
+      url: git@github.com:myOrg/myRepo.git
+      commit: 1234567890abcdef
+  someSource2:
+    git:
+      # This uses an HTTPS style git URL.
+      url: https://github.com/myOrg/myRepo.git
+      commit: 1234567890abcdef
+      # keepGitDir: true # Keep the .git directory when fetching the git source
```

Comment:
Something like this maybe so it gets picked up by syntax highlighter? 
```suggestion
    git:
      # This uses an HTTPS style git URL.
      url: https://github.com/myOrg/myRepo.git
      commit: 1234567890abcdef
      keepGitDir: true # [Optional] Keep the .git directory when fetching the git source. Default value: false
```

Reply:
Fixed.